samples/sobel/sobel_main.py

```diff
@@ -1,56 +1,231 @@
+|-----------+--------------+-----------+---------------------+---------------+------------------|
+| B_x       |          400 |   2240800 |                5602 |           N/A |              N/A |
+| + B_y     |          400 |      5600 |                  14 |           N/A |              N/A |
+| E_x1      |            3 |       123 |                  41 |           N/A |              N/A |
+| + E_y1    |            3 |        39 |                  13 |           N/A |              N/A |
+| ++ E_ra2  |          398 |  20751720 |               52140 |           N/A |              N/A |
+| +++ E_ra3 |          398 |     52138 |                 131 |           N/A |              N/A |
+| D_x3      |            3 |       123 |                  41 |           N/A |              N/A |
+| + D_y2    |            3 |        39 |                  13 |           N/A |              N/A |
+| ++ D_ra0  |          398 |  20751720 |               52140 |           N/A |              N/A |
+| +++ D_ra1 |          398 |     52138 |                 131 |           N/A |              N/A |
+| Fimg_x5   |          398 |   4436108 |               11146 |           N/A |              N/A |
+| + Fimg_y3 |          398 |     11144 |                  28 |           N/A |              N/A |
++-----------+--------------+-----------+---------------------+---------------+------------------+
+* Units in clock cycles
+
+With Optimization:
++-----------------------+--------------+-----------+---------------------+---------------+------------------+
+|                       |   Trip Count |   Latency |   Iteration Latency |   Pipeline II |   Pipeline Depth |
+|-----------------------+--------------+-----------+---------------------+---------------+------------------|
+| B_x_B_y               |       160000 |    320012 |                 N/A |             2 |               15 |
+| E_x_reuse_E_y_reuse   |       160000 |    160121 |                 N/A |             1 |              123 |
+| D_x_reuse1_D_y_reuse1 |       160000 |    160121 |                 N/A |             1 |              123 |
+| Fimg_x3_Fimg_y1       |       158404 |    158431 |                 N/A |             1 |               29 |
++-----------------------+--------------+-----------+---------------------+---------------+------------------+
+* Units in clock cycles
+"""
 
-f(hcl_A, hcl_Gx,hcl_Gy, hcl_F)
-npF = hcl_F.asnumpy()
+# The displayer also supports querying of different parts of the report, be it
+# with loop names and/or latency categories. For instance, if you want to query
+# the 'Latency' and 'Pipeline II' information of loops 'B' and 'E', we can tell
+# the displayer to query only that information. Since it can support multiple
+# queries, the arguments must be in a form of a list. It can also take in the
+# information regarding the loop-nest depths (loop level) in the program.
+report.display(loops=['B', 'E'], cols=['Latency', 'Pipeline II'])
 
-newimg = np.zeros((height-2,width-2,3))
```

Comment:
We should keep the verification part. To make sure the output from HCL is same as the ground truth.